examples/attitude_example.cpp

```diff
@@ -0,0 +1,150 @@
+   * @note Uses a modified world offset rotation of \f$\pi\f$ radians about the
+   *       roll axis
+   * @note This is is a shared pointer to a continually updated `Attitude` 
+   *       object.  There is no need to update in the contorl loop itself.
+   */
+  std::shared_ptr<kodlab::Attitude<float>> att_;
+
+  /**
+   * @brief Read-only attitude object for storing IMU data
+   * @note Uses a modified world offset rotation of \f$\pi\f$ radians about the
+   *       roll axis
+   * @note Unlike `att_` above which is a shared pointer to a continually 
+   *       updated `Attitude` object, this object is only updated when 
+   *       specifically assigned.
+   */
+  kodlab::Attitude<float> att_read_only_;
+
+  /**
+   * @brief Gathers attitude data and sets command torques to zero
+   */
+  void CalcTorques() override
+  {
+    // Get Updated Read-Only Attitude
+    att_read_only_ = robot_->GetAttitude();
+    
+    // Get Updated Attitude Representations
+    Eigen::Quaternionf quat = att_->get_att_quat();
+    kodlab::rotations::EulerAngles euler = att_->get_att_euler();
+    Eigen::Matrix3f rot_mat = att_->get_att_rot_mat();
+
+    // Print Attitude Representations to Terminal
+    std::cout << "att_:" << std::endl;
+    att_->PrintAttitude();
+    std::cout << "att_read_only_:" << std::endl;
+    att_read_only_.PrintAttitude();
+    std::cout << std::endl;
+
+    // Set Torques to Zero
+    std::vector<float> torques(num_motors_, 0);
+    robot_->SetTorques(torques);
+  }
+
+  /**
+   * @brief Prepares log
+   */
+  void PrepareLog() override
+  {
+    // Attitude Quaternion
+    log_data_.attitude_quaternion[0] = att_->get_att_quat().x();
+    log_data_.attitude_quaternion[1] = att_->get_att_quat().y();
+    log_data_.attitude_quaternion[2] = att_->get_att_quat().z();
+    log_data_.attitude_quaternion[3] = att_->get_att_quat().w();
+
+    // Attitude Euler Angles
+    log_data_.attitude_euler[0] = att_->get_att_euler().roll;
+    log_data_.attitude_euler[1] = att_->get_att_euler().pitch;
+    log_data_.attitude_euler[2] = att_->get_att_euler().yaw;
+
+    // Angular Velocity
+    log_data_.angular_velocity[0] = att_->get_ang_rate().x();
+    log_data_.angular_velocity[1] = att_->get_ang_rate().y();
+    log_data_.angular_velocity[2] = att_->get_ang_rate().z();
+
+    // Linear Acceleration
+    log_data_.linear_acceleration[0] = att_->get_accel().x();
+    log_data_.linear_acceleration[1] = att_->get_accel().y();
+    log_data_.linear_acceleration[2] = att_->get_accel().z();
+  }
+
+};
+
+int main(int argc, char **argv)
+{
+  // Setup Joints
+  std::vector<kodlab::mjbots::JointMoteus> joints;
+  joints.emplace_back(100, 4, 1, -1.3635165, 1, 1);
+  joints.emplace_back(101, 4, -1, 2.688, 5.0 / 3.0, 1);
+  joints.emplace_back(108, 4, 1, -0.4674585, 1, 1);
+
+  // Define Robot Options
+  kodlab::mjbots::ControlLoopOptions options;
+  options.log_channel_name = "attitude_data";
```

Comment:
Maybe world_offset should optionally be set here

Reply:
Agreed, this option would be nice to have.

Reply:
Feature implemented in 27d831239563df30c894c879e6625264fb52358a